src/gui/updater/nourlwidget.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 by Fabian Müller <fmueller@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QScopedPointer>
+#include <QWidget>
+
+namespace Ui {
+class Ui_NoUrlWidget;
+}
+
+namespace OCC {
+
+class NoUrlWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    explicit NoUrlWidget(QWidget *parent, const QString &statusMessage);
```

Comment:
What is the meaning of the class name?

Reply:
It comes from the method that previously showed the dialog: `WindowsUpdater::showNoUrlDialog`. It's called from `WindowsUpdater::versionInfoArrived`, when the info has an empty URL. 🤷‍♂️

Reply:
Rename to `NewVersionAvailableDialog`, same for the calling method.